hack/generator/pkg/astmodel/file_writer.go

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package astmodel
+
+import (
+	"bufio"
+	"io"
+	"os"
+
+	"github.com/dave/dst"
+	"github.com/dave/dst/decorator"
+)
+
+// GoSourceFile represents a file of Go code
+type GoSourceFile interface {
```

Comment:
Should the file be called `go_source_file.go`? #nitpick

Reply:
Fixed